### DISCUSSION TITLE
Twitter web hook

### DISCUSSION THREAD

**Discussion by jweston010** [AzuraCast/AzuraCast#4255] [Q&A] [answered]
I have my Twitter web hook set to tweet when a  DJ connects and disconnects. Along with station going on and off line. Sometimes, when a DJ connects, it tweets out some of the songs that the DJ is playing, but not all. I wish this behavior was more consistent, so that it would behave like The tweet every time a song changes behavior. So that when DJs are live,  it would tweet each song the dj plays, in case a particular song catches the listeners attention. Of course, when automation is on it doesn't need to tweet.￼

**Comment by BusterNeece** [comment]
@jweston010 We don't currently have any setting that lets you tweet new tracks only when a DJ is live. The "any time a song changes" setting is entirely separate from the "DJ connects/disconnects" setting.

**Comment by jweston010** [comment]
I know that the settings are separate, I was just thinking how useful 
it might be if those 2 functions could be combined to do what I mentioned.
I do wonder what causes azuracast to in fact twet some songs when a DJ 
is live when the tweet every song setting is not enabled?
I had the tweet every song check box checked, but heard from twitter 
followers that it was annoying so I turned it off.
I mean I could run the web hook test during say my live show every 
song, to manually have it tweet but that'd mean reloading the page like 
33 times in a 4 hour period.
This was just an idea I had.

**Reply by BusterNeece** [reply]
@jweston010 Oh I see what you mean. Yes, I can see how combining the two would be useful in some circumstances; it might be worthwhile to submit that over on https://features.azuracast.com/ for possible future inclusion?

**Comment by jweston010** [comment, marked as answer]
I'll do that. Thanks.
